lib/appSession.js

```diff
@@ -84,29 +84,34 @@ module.exports = (config) => {
   }
 
   function setCookie(
-    req,
```

Comment:
What I had in my head was keeping the signature the same, then for `req` we'd expect a https://nodejs.org/api/http.html#http_request_protocol so we couldn't rely on things like https://expressjs.com/en/api.html#req.secure but rather https://nodejs.org/api/http.html#http_request_protocol

Reply:
Would keeping the signature the same be useful if we are only using it for the secure flag (or well protocol)? All the other bits that used to refer `req`, don't refer to it anymore as we didn't want to be adding things on to the request object.

The reason why I changed the signature is:

- It is a private method, so didnt need to worry about backwards compatibility
- All we needed was a boolean, not the entire request object.

That said, I don't mind bringing back the request object, but that still requires a different signature as we need something to store the session and cookies in, as well as way to add and clear the cookies.

We can use the NodeJS way to set and clear cookies, but that was a bit more code and I felt like it might be nice to leverage framework specific cookie wrappers by allowing them to be injected.